deepmd/dpmodel/output_def.py

```diff
@@ -289,21 +293,69 @@ def get_deriv_name(name: str) -> Tuple[str, str]:
     return name + "_derv_r", name + "_derv_c"
 
 
+def apply_operation(var_def: OutputVariableDef, op: OutputVariableOperation) -> int:
+    """Apply a operation to the category of a variable definition.
+
+    Parameters
+    ----------
+    var_def : OutputVariableDef
+        The variable definition.
+    op : OutputVariableOperation
+        The operation to be applied.
+
+    Returns
+    -------
+    int
+        The new category of the variable definition.
+    """
+    return var_def.category | op.value
+
+
+def check_operation_applied(
```

Comment:
I am confused by the behavior of the function. could you please provide uts so I can better understand?